### PR TITLE
Feature/flow dps access merged with DPS BDS proxy

### PR DESCRIPTION
**Pull request by miklosszegedi**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,32 +14,91 @@ FROM build-setup AS build-dependencies
...
-FROM scratch as production
+FROM build-env as production
+
+WORKDIR /app/src
+COPY --from=build-env /app/main /app/main
+
+CMD ["go", "run", "-tags=relic", "cmd/api-service/main.go"]
+
+## (6) Add the statically linked binary to a distroless image
+FROM golang:1.17 as production-small
```
Comment (franklywatson) [inline Dockerfile]: FYI We've moved up to Golang 1.18 now

File: src/cmd/flow-dps/server.go [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/encoding/json"
+	"github.com/onflow/flow-go/engine/common/rpc/convert"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow/protobuf/go/flow/access"
+	"github.com/onflow/flow/protobuf/go/flow/entities"
+
+	conv "github.com/optakt/flow-dps/models/convert"
```
Comment (franklywatson) [inline src/cmd/flow-dps/server.go:18]: I think we should depend on the dapperlabs/flow-dps rather than the origin Optakt repo

File: src/go.mod [hunk 3]
```diff
@@ -1,202 +1,101 @@
...
+//replace github.com/onflow/api-service => github.com/GetElastech
+
 go 1.18
 
 require (
+	github.com/golang/protobuf v1.5.2
+	github.com/onflow/cadence v0.21.3-0.20220513161637-08b93d4bb7b9
 	github.com/onflow/flow-go v0.26.0
+	github.com/onflow/flow/protobuf/go/flow v0.2.5
+	github.com/optakt/flow-dps v1.4.8
```
Comment (franklywatson) [inline src/go.mod:13]: Let's stick to using the dapperlabs flow-dps repo. Thanks

File: Dockerfile (hunk 1 above)
Comment (miklosszegedi) [reply Dockerfile]: Addressed: https://github.com/GetElastech/api-service/commit/f6de78ce82434034e3e84702b21fb96a7a4e81d4

File: src/cmd/flow-dps/server.go (hunk 2 above)
Comment (A-town21) [reply src/cmd/flow-dps/server.go:18]: The only problem is that dapperlabs/flow-dps uses `module github.com/optakt/flow-dps` as it's import path. Go will not allow use of these packages to be imported using `dapperlabs/flow-dps` as is. Please see https://github.com/dapperlabs/flow-dps/blob/master/go.mod

**Comment by miklosszegedi** [comment]
This is now continued in https://github.com/GetElastech/api-service/pull/9

**Pull request closed without merging**